spring-cloud-data-module-deployers/spring-cloud-data-module-deployer-spi/src/main/java/org/springframework/cloud/data/module/deployer/ModuleDeployer.java

```diff
@@ -32,45 +34,45 @@
 public interface ModuleDeployer {
 
 	/**
-	 * Deploy the given {@code ModuleDescriptor}. Implementations
+	 * Execute the given {@code ModuleDeploymentRequest}. Implementations
 	 * may perform this operation asynchronously; therefore
 	 * a successful deployment may not be assumed upon return.
 	 * To determine the status of a deployment, invoke
-	 * {@link #status(org.springframework.xd.module.ModuleDescriptor.Key)}.
+	 * {@link #status(ModuleKey)}.
 	 *
-	 * @param descriptor descriptor for module to be deployed
+	 * @param request request for module to be deployed
 	 *
 	 * @throws IllegalStateException if the module has already been deployed
 	 */
-	void deploy(ModuleDescriptor descriptor);
+	void deploy(ModuleDeploymentRequest request);
```

Comment:
the JavaDoc could state something like: "returns a ModuleDeploymentId that can be used to check on the status since the deployment is triggered asynchronously"

I would view it like a POST operation returning a URL in the location header